.github/workflows/test_and_lint.yml

```diff
@@ -0,0 +1,78 @@
+name: test and lint
+
+on:
+  push:
+    branches: [ github-actions ]
+  pull_request:
+    branches: [ github-actions ]
+
+jobs:
+
+  build-and-test:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        node-version: [10.x]
+
+    steps:
+    - name: install dependencies on ubuntu
+      if: startsWith(matrix.os,'ubuntu')
+      run: |
+        sudo apt install -y make gcc pkg-config build-essential libx11-dev libxkbfile-dev 
+    
+    - name: install dependencies on windows
+      if: startsWith(matrix.os,'windows')
+      run: |
+        npm --add-python-to-path install --global --production windows-build-tools
+      shell: powershell 
+    
+    - uses: actions/checkout@v2
+    
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    
+    - name: Cache node_modules with yarn
+      uses: actions/cache@v2
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    
+    - run: yarn --frozen-lockfile
+    
+    - run: yarn test
+      env:
+        CI: true
+        
+  code-lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out Git repository
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 10.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 10
+
+    # ESLint and Prettier must be in `package.json`
+    - run: yarn --frozen-lockfile
```

Comment:
It is to prevent the CI server update the minor version of dependencies. Imho it helps to avoid unexpected behaviours.

Imagine that you install some packages, then finish the dev and push it. In the mean time, a minor version is up, then now the build server will actually run on different versions of dependencies with your machine. If something breaks it can be tricky to know by then.

Reply:
Not sure about this one. It might not do us much good to fail the build because a 5th level dependency was unpublished and we're not allowing `yarn` to pick another version that's within the allowed dependency version range. The only difference is that it will force us to immediately fix `yarn.lock` before anything else can be merged. 

Also remember that our `yarn.lock` file will do nothing for other applications that eventually consume the trace extension.

Reply:
@marechal-p @vince-fugnitto  WDYT?

Reply:
I think that it's fine because this should mostly catch issues where someone bumped a version range in this repo but didn't commit the updated yarn.lock file. I think transitive dependencies are only updated when doing `yarn upgrade`, and maybe when something is unpublished like you mentioned but I don't know for sure. I'm ok with this flag being used, it can be changed later if there's issue with it but I'd try it out first. Unless enough people feel uncomfortable with it.

Reply:
Ok, we can give this a try - It's easily changed if it causes more problems than it solves.